EIPS/eip-4444.md

```diff
@@ -0,0 +1,101 @@
+---
+eip: 4444
+title: Bound Historical Data in Execution Clients
+description: Prune historical data in clients older than one year
+author: George Kadianakis (@asn_d6), lightclient (@lightclient), Alex Stokes (@ralexstokes)
+discussions-to: TBD
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-11-02
+---
+
+## Abstract
+
+Clients must stop serving historical blocks/receipts that are older than a year on the p2p layer. Clients may prune such historical data.
```

Comment:
**MUST** seems overly prescriptive here.  Perhaps instead we could say **MAY**?

Reply:
Hello. The idea here is that if we make this optional a client Alice might continue serving old data, and Bob will rely on Alice out of inertia. At some point, it's likely that Bob will get rugged by Alice since this is an optional feature and Alice might get bored of serving old data. By making this a **MUST** it forces Bob to find a long-term solution to his problem.

Reply:
I can see **SHOULD** being applied along with rationale for it (though I don't agree with the rationale).  **MUST** should be used only if it is an error to not follow the rules.  In this case, it is not an error to serve old data so it shouldn't be a **MUST**.

Reply:
Note, that discussions with @karalabe have pointed to making this a **MUST** because otherwise users will still rely on it while quality of this feature degrades until it is unusable. A **MUST**, instead, forces users and dapps to actually "upgrade" how they utilize this at the point of this shipping, forcing everyone's hand so it doesn't silently get worse and worse.

If we go that path, then this spec should specify that devp2p *does* return errors on requests outside of the specified range of epochs/time

Reply:
Moved to magicians thread.

Reply:
A new update on this but TheGraph just censored Tornado Cash.
The underlying problem is things like databased shared over ipfs or Google Bigquery is only for static content so wouldn’t allows us to have a realtime service as it would take a day to download updates. We also need very old data in order to let a user withdraw funds deposited several years ago.

This isn’t that much a problem currently as this just mean we will need our own home (not cloud) hosted node with parity_tracing, along for getting smart contract storage range at past blocks. Such thing is possible because full anciant data is broadcasted and updated in real time but show why it is important to have node/rpc be able to broadcast full chain history over the p2p network.
**Yet at the same time, it is important for censorship resistance to not have the protocol upgradable**

If third party service are made required, then well, Ethereum, will be decentralized at the currency level like Bitcoin while **Dapps like casinos or yield farming, will be fully permissionned have to register with authorities by paying an army of lawayers in order to be allowed to run which means, DeFi won’t be that much different than Fintechs on the tradionnal banking system which rent their computing hardware.**

I think keeping history on the p2p definitely worth the reduced transactions per seconds outputs or if we decide to behave like swift or MasterCard or Visa for being able to run as fast as them.